EDU-1088: clarify partial string match information (#2311)
temporalio/documentation: docs-src/concepts/what-is-a-list-filter.md

### Partial string match

> The `=` operator functions similarly to **CONTAINS**, helping to find Workflows with Search Attributes containing a specific word.

Reviewer: Because "function" has such a specific meaning in software development, I'm inclined to use "works" in this context. And although "like" provides an informal tone, it is indeed accepted by the Google style guide.

> Partial string matching only applies to locating Text Search Attributes.

Reviewer: ```suggestion
Partial string matching applies only to locating Text Search Attributes.
```

<!-- note: advanced vis features will be supported in SQL upon the release of v1.20.-->

> For instance, if you have a custom Search Attribute named `Description` of `Text` type with the value "The quick brown fox jumps over the lazy dog", a search for `Description='quick'` or `Description='fox'` will successfully return the Workflow.

Reviewer: ```suggestion
For example, if you have a custom Search Attribute named `Description` of `Text` type with the value "The quick brown fox jumps over the lazy dog", a search for `Description='quick'` or `Description='fox'` successfully returns the Workflow.
```
Per https://developers.google.com/style/word-list#for-instance.

However, searches for partial words like `Description='qui'` or `Description='laz'` won't return the Workflow.
This limitation arises because [Elasticsearch's tokenizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-standard-tokenizer.html) is configured to return complete words as tokens.

### Efficient API usage

> If the Advanced List Filter API retrieves a substantial number of Workflow Executions (over 10,000), the response time might be longer.

Reviewer: ```suggestion
If the Advanced List Filter API retrieves a substantial number of Workflow Executions (more than 10,000), the response time might be longer.
```

> Beginning with Temporal Server v1.20 you can employ the `CountWorkflow` API to efficiently count the number of [Workflow Executions](/concepts/what-is-a-workflow-execution).

Reviewer: ```suggestion
Beginning with Temporal Server v1.20, you can employ the `CountWorkflow` API to efficiently count the number of [Workflow Executions](/concepts/what-is-a-workflow-execution).
```

> Starting from Temporal Server v1.20 and later, you can employ the `CountWorkflow` API to efficiently count the number of [Workflow Executions](/concepts/what-is-a-workflow-execution).

Reviewer: I recommend stetting the original.

If keeping "Starting from", delete "and later". And because the reader isn't "starting" Temporal Server in this context, I recommend "Beginning with".

To paginate the results using the `ListWorkflow` API, use the page token to retrieve the next page.
Continue until the page token becomes `null`/`nil`.

#### List Filter examples

> Here are examples of List Filters set with [`tctl`](/tctl-v1/workflow/list):

Reviewer: If you're referring to this entire section, change the ending colon to a period to avoid implying that the following code block contains multiple examples.

Author: note to self: remember to change this once `tctl` is officially deprecated

```
WorkflowType = "main.YourWorkflowDefinition" and ExecutionStatus != "Running" and (StartTime > "2021-06-07T16:46:34.236-08:00" or CloseTime > "2021-06-07T16:46:34-08:00")
```

> When used, you will receive a list of Workflows fulfilling the following criteria:

Reviewer: ```suggestion
When you use the preceding example, you receive a list of Workflows fulfilling the following criteria:
```

- Workflow Type is `main.YourWorkflowDefinition`.
- Workflow isn't in a running state.
- Workflow either started after "2021-06-07T16:46:34.236-08:00" or closed after "2021-06-07T16:46:34-08:00".

> Additional examples of List Filters are provided below.

Reviewer: ```suggestion
The following are additional examples of List Filters.
```

### Partial string match

The `=` operator works like **CONTAINS** to find Workflows with Search Attributes that contain a specific word.

> Partial string match can only be used to find `Text` Search Attributes.

Reviewer 2: ```suggestion
You can only employ partial string matching to locate Text Search Attributes.
```

passive -> active
